scaladoc/src/dotty/tools/scaladoc/SocialLinks.scala

```diff
@@ -0,0 +1,25 @@
+package dotty.tools.scaladoc
+
+import java.nio.file.Path
+import java.nio.file.Paths
+import dotty.tools.dotc.core.Contexts.Context
+
+enum SocialLinks(val url: String, val whiteIconName: String, val blackIconName: String):
+  case Github(ghUrl: String) extends SocialLinks(ghUrl, "github-icon-white.png", "github-icon-black.png")
+  case Twitter(tUrl: String) extends SocialLinks(tUrl, "twitter-icon-white.png", "twitter-icon-black.png")
+  case Gitter(gUrl: String) extends SocialLinks(gUrl, "gitter-icon-white.png", "gitter-icon-black.png")
+  case Discord(dUrl: String) extends SocialLinks(dUrl, "discord-icon-white.png", "discord-icon-black.png")
+  case Custom(cUrl: String, cWhiteIconName: String, cBlackIconName: String) extends SocialLinks(cUrl, cWhiteIconName, cBlackIconName)
+
+object SocialLinks:
+  def parse(s: String): Either[String, SocialLinks] =
+    val splitted = s.split("::")
+    if splitted.size < 2 then Left(s"Social links arg $s is invalid.") else
```

Comment:
Can we add more meaning full error messages? e.g. when `custom` is provided then we should say that there is incorrect number of arguments. We should also check in e.g. `github` is provided that there are no additional arguments